src/model-viewer-base.ts

```diff
@@ -211,11 +213,24 @@ export default class ModelViewerElementBase extends UpdatingElement {
     }
 
     if (HAS_INTERSECTION_OBSERVER) {
+      const enterRenderTreeProgress = this[$progressTracker].beginActivity();
+
       this[$intersectionObserver] = new IntersectionObserver(entries => {
         for (let entry of entries) {
           if (entry.target === this) {
-            this[$scene].isVisible = entry.isIntersecting;
-            this.requestUpdate();
+            const oldValue = this[$isInRenderTree];
+            this[$isInRenderTree] = this[$scene].isVisible =
```

Comment:
Can we remove this[$scene].isVisible? It seems to be duplicate.

Reply:
No, I think it should stay. The reason why is that it controls the notion of visibility on the Three.js side, while `$isInRenderTree` is a detail of the DOM element. If we relied only `$isInRenderTree`, we would be building-in more dependencies on a DOM node in `Renderer`. And then, when we try to move Three.js stuff into a Worker, we have more work to separate those things back out.

Reply:
Fair, but ModelScene extends Scene, which already has a visible member. Can we just use that instead of creating isVisible on top of it? Or do they still have separate meanings?

Reply:
That's a good point. My guess is that we can re-use `scene.visible`. I'll look into it.

Reply:
Switched over. AFAIK `scene.visible` will always be `true` in the code path where we actually render, so it should be fine.